### ISSUE TITLE
`stop-iteration-return` false positive when using `itertools.cycle`

### ISSUE THREAD

**Issue by richardfearn** [PyCQA/pylint#7765]
### Bug description

Running pylint on this code gives a false positive:

```
from itertools import cycle


def test():
    vals = cycle('ABCD')
    while True:
        yield next(vals)
```

This is similar to #2158 which was fixed by adding a list of [known infinite iterators](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/refactoring/refactoring_checker.py#L38).

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:7:14: R1708: Do not raise StopIteration in generator, use return statement instead (stop-iteration-return)
```

### Expected behavior

No warning, because `cycle` [repeats indefinitely](https://docs.python.org/3/library/itertools.html#itertools.cycle).

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.7 (main, Nov  2 2022, 18:49:29) [GCC 12.2.0]
```

### OS / Environment

Ubuntu 22.10

**Comment by richardfearn** [comment]
Thanks for the quick fix!
